### PR TITLE
Add PROXY protocol support to `Server`

### PR DESCRIPTION
**Pull request by drclau**
**What this PR does**:
Adds support for PROXY protocol to `Server`, by using `go-proxyproto` package.

**Which issue(s) this PR fixes**:

Fixes N/A

**Checklist**
- [x] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -592,3 +599,13 @@ func (s *Server) Shutdown() {
 	_ = s.HTTPServer.Shutdown(ctx)
 	s.GRPC.GracefulStop()
 }
+
+func newProxyProtocolListener(httpListener net.Listener, readHeaderTimeout time.Duration) net.Listener {
+	// Wraps the listener with a proxy protocol listener.
+	// NOTE: go-proxyproto supports non-PROXY, PROXY v1 and PROXY v2 protocols via the same listener.
+	// Therefore, enabling this feature does not break existing setups.
```
Comment (grafanalf) [inline server/server.go:606]: I suggest you move this comment into a more visible place. Perhaps into the README.md at the root of the repository?

File: server/server_test.go [hunk 2]
```diff
@@ -723,36 +740,72 @@ func (f *FakeLogger) assertNotContains(t *testing.T, content string) {
...
+		require.NoError(t, err)
+
+		logger.assertContains(t, "sourceIPs=127.0.0.1")
 	})
 
-	go func() {
-		require.NoError(t, server.Run())
-	}()
-	defer server.Shutdown()
+	t.Run("without PROXY protocol", func(t *testing.T) {
```
Comment (grafanalf) [inline server/server_test.go:784]: The name of the test is wrong: should be `with PROXY protocol` (not without) as per line 787.

File: server/server.go (hunk 1 above)
Comment (drclau) [reply server/server.go:606]: I like your suggestion. Though, the `README.md` at the root does not seem to be the right place, considering how generic it is and how little information it contains. What do you think about creating a new `README.md` in the `server` directory instead?
I noticed that other directories do have their own `README`s.

Comment (grafanalf) [reply server/server.go:606]: Sounds great.

**Review by aarongodin — APPROVED** [review]
Beyond the suggestions already made, LGTM

File: server/server.go (hunk 1 above)
Comment (drclau) [reply server/server.go:606]: I went for adding `PROXYPROTOCOL.md`, because the information is very specific.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**